src/test/rpc_tests.cpp

```diff
@@ -308,4 +343,205 @@ BOOST_AUTO_TEST_CASE(rpc_ban)
     BOOST_CHECK_EQUAL(adr.get_str(), "2001:4d48:ac57:400:cacf:e9ff:fe1d:9c63/128");
 }
 
+
+BOOST_AUTO_TEST_CASE(rpc_getblockchaininfo)
+{
+    // test bip109 counting
+
+    // need the chain params for setting up test chains and cases
+    SelectParams(CBaseChainParams::MAIN);
+    const Consensus::Params& params = Params().GetConsensus();
+    // we need 2 more than the size of the majority window because
+    // the last block (tip) does not count (ref. "blocks" entry
+    // listing height as 1 less than the size of the chain)
+    //const uint32_t chain_len = params.nMajorityWindow + 1;
+    const int chain_len = params.nMajorityWindow + 1;
+    
+    // ensure that the hard fork is not yet seen as active
+    // otherwise the global pblocktree must be set up right,
+    // or else an assert will trip in HardForkMajorityDesc
+    sizeForkTime.store(std::numeric_limits<uint32_t>::max());
+    
+    ///////////////////////////////////////////////////////////////////////////
+    // case 1: a block history with only version 4 blocks (zero bip109 found)
+    std::vector<CBlockIndex> *testblocks;
+    testblocks = freshchain(chain_len);
```

Comment:
I guess we don't need chain_len everywhere, just 3 or so would hit all the code and actually test what we want to test, no?